plugin/documents.py

```diff
@@ -149,13 +171,27 @@ def on_hover(self, point: int, hover_zone: int) -> None:
             return
         self.view.run_command("lsp_hover", {"point": point})
 
+    # --- textDocument/documentColor -----------------------------------------------------------------------------------
+
+    def _do_color_boxes_async(self) -> None:
+        session = self.session("colorProvider")
+        if session:
+            session.send_request(Request.documentColor(document_color_params(self.view)), self._on_color_boxes)
+
+    def _on_color_boxes(self, response: Any) -> None:
```

Comment:
This also runs on async thread (not ST's own async thread though).

Reply:
Yeah. I don't know what to do with these kinds of callbacks. I'm thinking of also running all response handlers on ST's async thread to obtain a more serialized state of the plugin. Because if everything runs on ST's async thread, we don't really need locks anymore.